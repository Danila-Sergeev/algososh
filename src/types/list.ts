import { ElementStates } from "./element-states";
import { ListNode } from "../components/list-page/list-node";

export type LinkedArrElement = { value: string; state: ElementStates };
export type RemovingNode = { id: string; value: string } | null;

export interface ILinkedList<T extends LinkedArrElement> {
  head: ListNode<T> | null;
  tail: ListNode<T> | null;

  addToHead(value: T): void;

  addToTail(value: T): void;

  removeHead(): T | null;

  removeTail(): T | null;

  toArray(): ListNode<T>[];

  insertAtIndex(value: T, index: number): void;

  removeAtIndex(index: number): LinkedArrElement | null;

  getLength(): number;

  clearNodeValue(index: number): void;

  getNodeAtIndex(index: number): ListNode<T> | null;

  changeState(index: number, state: ElementStates): void;

  getData(): {
    array: ListNode<T>[];
    tail: ListNode<T> | null;
    head: ListNode<T> | null;
  };
}
