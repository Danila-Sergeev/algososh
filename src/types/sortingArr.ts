import { ElementStates } from './element-states';

export type ArrayElement = { value: number; state: ElementStates };
