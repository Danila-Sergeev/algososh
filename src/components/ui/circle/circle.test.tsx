import React from "react";
import { render } from "@testing-library/react";
import { Circle } from "./circle"; // Измените путь к вашему компоненту
import { ElementStates } from "../../../types/element-states";

describe("Circle Component", () => {
    it("рендерит элемент без буквы", () => {
        const { asFragment } = render(<Circle />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("рендерит элемент с буквой", () => {
        const { asFragment } = render(<Circle letter="A" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("рендерит с head", () => {
        const { asFragment } = render(<Circle head="Test Head" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("рендерит с react-элементом в head", () => {
        const { asFragment } = render(
            <Circle head={<span>React Element</span>} />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("рендерит с tail", () => {
        const { asFragment } = render(<Circle tail="Tail Text" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("рендерит с react-элементом в tail", () => {
        const { asFragment } = render(
            <Circle tail={<span>Tail Element</span>} />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("рендерит с index", () => {
        const { asFragment } = render(<Circle index={5} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("рендерит с пропом isSmall === true", () => {
        const { asFragment } = render(<Circle isSmall />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("рендерит в состоянии default", () => {
        const { asFragment } = render(<Circle state={ElementStates.Default} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("рендерит в состоянии changing", () => {
        const { asFragment } = render(
            <Circle state={ElementStates.Changing} />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("рендерит в состоянии modified", () => {
        const { asFragment } = render(
            <Circle state={ElementStates.Modified} />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
