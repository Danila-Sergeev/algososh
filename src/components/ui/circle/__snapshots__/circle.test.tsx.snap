// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle Component рендерит в состоянии changing 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-test-id="circleHead"
    />
    <div
      class="circle   changing"
      data-test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-test-id="circleTail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component рендерит в состоянии default 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-test-id="circleHead"
    />
    <div
      class="circle   default"
      data-test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-test-id="circleTail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component рендерит в состоянии modified 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-test-id="circleHead"
    />
    <div
      class="circle   modified"
      data-test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-test-id="circleTail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component рендерит с head 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
      data-test-id="circleHead"
    >
      Test Head
    </div>
    <div
      class="circle   default"
      data-test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-test-id="circleTail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component рендерит с index 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-test-id="circleHead"
    />
    <div
      class="circle   default"
      data-test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      5
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
      data-test-id="circleTail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component рендерит с react-элементом в head 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-test-id="circleHead"
    >
      <span>
        React Element
      </span>
    </div>
    <div
      class="circle   default"
      data-test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-test-id="circleTail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component рендерит с react-элементом в tail 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-test-id="circleHead"
    />
    <div
      class="circle   default"
      data-test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-test-id="circleTail"
    >
      <span>
        Tail Element
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Circle Component рендерит с tail 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-test-id="circleHead"
    />
    <div
      class="circle   default"
      data-test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
      data-test-id="circleTail"
    >
      Tail Text
    </div>
  </div>
</DocumentFragment>
`;

exports[`Circle Component рендерит с пропом isSmall === true 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-test-id="circleHead"
    />
    <div
      class="circle  small is-small default"
      data-test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-test-id="circleTail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component рендерит элемент без буквы 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-test-id="circleHead"
    />
    <div
      class="circle   default"
      data-test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-test-id="circleTail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle Component рендерит элемент с буквой 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-test-id="circleHead"
    />
    <div
      class="circle   default"
      data-test-id="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        A
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-test-id="circleTail"
    />
  </div>
</DocumentFragment>
`;
